### ISSUE TITLE
CS0721 error reported on unintuitive location

### ISSUE THREAD

**Issue by AlFasGD** [dotnet/roslyn#61831]
The [CS0721](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0721) error is reported on the name of the declared method instead of the identifier of the static type of the parameter.

## Example

**Current Behavior**:
```csharp
public void Method(Static1 static, Static2 static2) { }
            ~~~~~~
```

**Expected Behavior**:
```csharp
public void Method(Static1 static, Static2 static2) { }
                   ~~~~~~~         ~~~~~~~
```

With this change, the error message would also have to remove the name of the parameter whose type is static:
```diff
- 'type': static types cannot be used as parameters
+ Static types cannot be used as parameters
```

**Comment by RikkiGibson** [comment]
It feels like there's no need to change the diagnostic message. Let's just use a better location for the error.

It's interesting that a static type argument is reported on the parameter name. [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQAwAIEEYAsBuAsAKAIQGY10A2NAJhQGEUBvAlFtUjShTFAWQAoAMlADOAFwA8tAHwoAxugCUjZqwC+BVUA===).

For this issue, I think it would be fine to report on the parameter name here, but reporting on the offending type itself would be ideal.

**Comment by jjonescz** [comment]
Lambdas are not addressed, pending the merge of `feature/lambda-default-parameters`, see https://github.com/dotnet/roslyn/pull/63975#discussion_r971083738
